src/backend/looped/weapons/rapid_fire.cpp

```diff
@@ -14,50 +14,61 @@ namespace big
 
-					Vector3 end = camera_position + camera_direction * 2000.0;
+				Vector3 end = camera_position + camera_direction * 2000.0;
+				Hash weapon_hash;
+				WEAPON::GET_CURRENT_PED_WEAPON(self::ped, &weapon_hash, false);
 
-					const auto raycast_handle =
-					    SHAPETEST::START_EXPENSIVE_SYNCHRONOUS_SHAPE_TEST_LOS_PROBE(camera_position.x,
-					        camera_position.y,
-					        camera_position.z,
-					        end.x,
-					        end.y,
-					        end.z,
-					        -1,
-					        0,
-					        7);
-					int did_raycast_hit = 0;
-					Vector3 raycast_hit_position{};
-					Vector3 raycast_surface_normal_hit_position{};
-					Entity raycast_hit_entity{};
-					SHAPETEST::GET_SHAPE_TEST_RESULT(raycast_handle, &did_raycast_hit, &raycast_hit_position, &raycast_surface_normal_hit_position, &raycast_hit_entity);
-					if (did_raycast_hit)
+				if (WEAPON::GET_WEAPONTYPE_GROUP(weapon_hash) == 1548507267)
+				{
+					if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_ATTACK))
 					{
-						end = raycast_hit_position;
-					}
+						const auto raycast_handle =
+						    SHAPETEST::START_EXPENSIVE_SYNCHRONOUS_SHAPE_TEST_LOS_PROBE(camera_position.x,
+						        camera_position.y,
+						        camera_position.z,
+						        end.x,
+						        end.y,
+						        end.z,
+						        -1,
+						        0,
+						        7);
+						int did_raycast_hit = 0;
+						Vector3 raycast_hit_position{};
+						Vector3 raycast_surface_normal_hit_position{};
+						Entity raycast_hit_entity{};
+						SHAPETEST::GET_SHAPE_TEST_RESULT(raycast_handle, &did_raycast_hit, &raycast_hit_position, &raycast_surface_normal_hit_position, &raycast_hit_entity);
+						if (did_raycast_hit)
+						{
+							end = raycast_hit_position;
+						}
 
-					Hash weapon_hash;
-					WEAPON::GET_CURRENT_PED_WEAPON(self::ped, &weapon_hash, false);
+						Vector3 dim_min;
+						Vector3 dim_max;
+						MISC::GET_MODEL_DIMENSIONS(ENTITY::GET_ENTITY_MODEL(weapon_entity), &dim_min, &dim_max);
+						const auto dimensions = (dim_max - dim_min) * 0.5f;
+						const auto weapon_position =
+						    ENTITY::GET_OFFSET_FROM_ENTITY_IN_WORLD_COORDS(weapon_entity, dimensions.x, 0, dimensions.z * 0.3f);
 
-					MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(weapon_position.x, weapon_position.y, weapon_position.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+						MISC::SHOOT_SINGLE_BULLET_BETWEEN_COORDS(weapon_position.x, weapon_position.y, weapon_position.z, end.x, end.y, end.z, WEAPON::GET_WEAPON_DAMAGE(weapon_hash, 0), true, weapon_hash, self::ped, true, false, -1.0);
+					}
 				}
+				else
+				{
+					if (PAD::IS_DISABLED_CONTROL_PRESSED(0, (int)ControllerInputs::INPUT_ATTACK) && PED::GET_PED_CONFIG_FLAG(self::ped, 78, 0))
+					{
+						PED::SET_PED_SHOOTS_AT_COORD(self::ped, end.x, end.y, end.z, true);
+						WEAPON::REFILL_AMMO_INSTANTLY(self::ped);
```

Comment:
Did the user ask for this?

Reply:
what user?

Reply:
The user of the cheat. You are basically reduplicating infinite clip which the cheat already has.